--- a/contrib/postgres_fdw/postgres_fdw.c
+++ b/contrib/postgres_fdw/postgres_fdw.c
@@ -451,6 +451,7 @@ static void merge_fdw_options(PgFdwRelationInfo *fpinfo,
                                  const PgFdwRelationInfo *fpinfo_o,
                                  const PgFdwRelationInfo *fpinfo_i);

+static bool postgresIsForeignScanParallelSafe(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte);

 /*
  * Foreign-data wrapper handler function: return a struct with pointers
@@ -505,6 +506,8 @@ postgres_fdw_handler(PG_FUNCTION_ARGS)
        /* Support functions for upper relation push-down */
        routine->GetForeignUpperPaths = postgresGetForeignUpperPaths;

+    routine->IsForeignScanParallelSafe = postgresIsForeignScanParallelSafe;
+
        PG_RETURN_POINTER(routine);
 }

@@ -5822,3 +5825,6 @@ find_em_expr_for_rel(EquivalenceClass *ec, RelOptInfo *rel)
        /* We didn't find any suitable equivalence class expression */
        return NULL;
 }
+static bool postgresIsForeignScanParallelSafe(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte) {
+    return true;
+}